guides/common/modules/proc_adding-a-google-gce-connection.adoc

```diff
@@ -5,31 +5,50 @@ Use this procedure to add Google Compute Engine (GCE) as a compute resource in {
 To use the CLI instead of the {ProjectWebUI}, see the xref:cli-Adding_a_Google_GCE_Connection_{context}[].
 
 .Procedure
-include::snip_create-and-add-google-gce-service-account-key.adoc[]
+. In Google GCE, generate a service account key in JSON format.
 . In the {ProjectWebUI}, navigate to *Infrastructure* > *Compute Resources* and click *Create Compute Resource*.
 . In the *Name* field, enter a name for the compute resource.
 . From the *Provider* list, select *Google*.
 . Optional: In the *Description* field, enter a description for the resource.
-. In the *Google Project ID* field, enter the project ID.
-. In the *Client Email* field, enter the client email.
-. In the *Certificate Path* field, enter the path to the service account key.
-For example, `/usr/share/foreman/_gce_key_.json`.
+. In the *JSON key* field, click *Choose File* and locate your service account key for upload from your local machine.
 . Click *Load Zones* to populate the list of zones from your GCE environment.
 . From the *Zone* list, select the GCE zone to use.
 . Click *Submit*.
 
 [id="cli-Adding_a_Google_GCE_Connection_{context}"]
 .CLI procedure
-include::snip_create-and-add-google-gce-service-account-key.adoc[]
+. In Google GCE, generate a service account key in JSON format.
+. Copy the file from your local machine to {ProjectServer}:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# scp gce_key.json root@{foreman-example-com}:/usr/share/foreman/gce_key.json
+----
+. On {ProjectServer}, change the owner for your service account key to the `foreman` user:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# chown foreman /usr/share/foreman/_gce_key_.json
+----
+. On {ProjectServer}, configure permissions for your service account key to ensure that the file is readable:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# chmod 0600 /usr/share/foreman/_gce_key_.json
+----
+. On {ProjectServer}, restore SELinux context for your service account key:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# restorecon -vv /usr/share/foreman/_gce_key_.json
```

Comment:
Users should not write files to `/usr/share/foreman`. This feels like configuration, which belongs in `/etc`.

Second: it's better to make it owned by `root:foreman` and use mode `0640`. That way the `foreman` user can only read it, and that's what we want.

Reply:
If users download it and they can remove it afterwards, perhaps root's home directory would be better?
Config in `/etc` feels more permanent.

Reply:
The home directory would not be readable by the Foreman user; there does appear to be a weird API where a file path on the server is given and where the user `foreman` will read it. That's why I asked @stejskalleos if we can send the contents over the API. That would actually work in a networked environment, which Hammer is supposed to work in.